streams/src/test/java/org/apache/kafka/streams/state/internals/CompositeReadOnlySessionStoreTest.java

```diff
@@ -111,10 +112,40 @@ public void shouldThrowInvalidStateStoreExceptionOnRebalance() throws Exception
         store.fetch("a");
     }
 
-    @Test(expected = InvalidStateStoreException.class)
-    public void shouldThrowInvalidStateStoreExceptionIfFetchThrows() throws Exception {
+    @Test
+    public void shouldThrowInvalidStateStoreExceptionIfUnderlyingSessionStoreFetchThrows() {
+        underlyingSessionStore.setOpen(false);
+        try {
+            underlyingSessionStore.fetch("key");
+            fail("Should have thrown InvalidStateStoreException with underlying session store");
+        } catch (InvalidStateStoreException e) { }
+    }
+
+    @Test
+    public void shouldThrowInvalidStateStoreExceptionIfSessionFetchThrows() {
         underlyingSessionStore.setOpen(false);
-        underlyingSessionStore.fetch("key");
+        try {
+            sessionStore.fetch("key");
+            fail("Should have thrown InvalidStateStoreException with session store");
+        } catch (InvalidStateStoreException e) { }
     }
 
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionIfFetchingNullKey() {
+        sessionStore.fetch(null);
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionIfPuttingNullSessionWindowKey() {
+        Windowed<String> sessionKey = new Windowed<>(null, new SessionWindow(0, 0));
+        try {
+            underlyingSessionStore.put(sessionKey, 1L);
+            fail("Should have thrown NullPointerException while setting windowed sessionKey to null");
+        } catch (NullPointerException e) { }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionIfPuttingNullKey() {
+        underlyingSessionStore.put(null, 1L);
```

Comment:
`sessionStore.put(..)`  - should also add a null check in `CachingSessionStore.put(..)`